confdgnmi/testtool/OpenConfig/01__interfaces.robot

```diff
@@ -0,0 +1,195 @@
+    # non-root namespace
+    interfaces/openconfig-interfaces:interface
+    interfaces/openconfig-interfaces:interface/
+    /interfaces/openconfig-interfaces:interface
+    /interfaces/openconfig-interfaces:interface/
+    # both namespaces
+    openconfig-interfaces:interfaces/openconfig-interfaces:interface
+    /openconfig-interfaces:interfaces/openconfig-interfaces:interface
+    openconfig-interfaces:interfaces/openconfig-interfaces:interface/
+    /openconfig-interfaces:interfaces/openconfig-interfaces:interface/
+
+Get "interfaces" with "encoding" parameter for all supported encodings
+    [Documentation]    Verify that ``GetRequest`` with ``path=interfaces`` parameter
+    ...                receives OK response for all/any of the supported Encoding values.
+    [Tags]  encoding
+
+List "/interfaces/interface" contains at least one record
+    [Documentation]    Verify that ``GetRequest`` with ``path=interfaces/interface`` parameter
+    ...                receives OK response and contains at least one or more items.
+    [Tags]  list
+    Verify Get of  /interfaces/interface
+    ${count}=  Get last updates count
+    Should Not Be Equal As Integers   ${count}  0
+
+Read existing "/interfaces/interface" list entries one by one
+    [Documentation]    Verify that ``GetRequest`` with ``path=interfaces/interface[name]``
+    ...                parameter receives OK response for all/any of the supported Encoding values.
+    [Tags]  list
+    Verify Get of    /interfaces/interface
+    ${updates}=  Last updates
+    Log  ${updates}
+
+Try reading non-existing list entry from "/interfaces/interface"
+    [Documentation]    Verify that ``GetRequest`` with ``path=interfaces/interface[non-existing-name]``
+    ...                parameter for non-existing key value receives error response.
+    [Tags]  negative  list
+    Add Path Parameter    /interfaces/interface[name=non-existing-interface]
+    Dispatch Get Request
+    Should Received Error Response
+
+Read "prefix=/interfaces", "path=interface[]" list entries one by one
+    [Documentation]    Verify that list can be iterated with separate prefix/path parameters.
+    [Tags]  prefix  list
+
+Read "/interfaces/interface[]/name" key leaf from existing entry
+    [Documentation]    Verify that key value can be read using Get request with path parameter.
+    [Tags]  key  list
+    ${response}=  Get data from  /interfaces/interface[name=${INTERFACE}]/name
+    Should Be Equal    ${INTERFACE}    TODO
+
+Get "CONFIG" response includes "config" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does include the read/write YANG "config" container.
+    [Tags]  config  container
+    Set Datatype To    CONFIG
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check Last Updates Include    config
+
+Get "CONFIG" response does not include "state" container
```

Comment:
on "backend" - it does reuse code pieces, on end-user test level, it would lead to very short parametrized phrases that are harder to ingest by non-developers IMHO...